test/unit/gateways/d_local_test.rb

```diff
@@ -32,6 +33,54 @@ def test_failed_purchase
     assert_equal '300', response.error_code
   end
 
+  def test_successful_card_save
+    @gateway.expects(:ssl_post).times(2).returns(successful_authorize_response, successful_void_response)
+
+    response = @gateway.verify(@credit_card, @options)
+    assert_success response
+
+    assert_equal 'D-15104-be03e883-3e6b-497d-840e-54c8b6209bc3', response.authorization
+    assert_equal 'CV-ecd897ac-5361-45a1-a407-aaab044ce87e', response.primary_response.params['card']['card_id']
+    assert response.test?
+  end
+
+  def test_failed_verify_during_card_save
+    @gateway.expects(:ssl_post).returns(failed_authorize_response)
+
+    response = @gateway.verify(@credit_card, @options)
+    assert_failure response
+    assert_equal '309', response.error_code
+  end
+
+  def test_failed_void_during_card_save_and_verification
```

Comment:
what does this test do?

Reply:
Since in the case of dlocal we make a verify call which authorizes a 1$ charge (not capture) with `card_save` set to true and voids it immediately. This check is to see even if the void action fails, we should be getting the generated `card.card_id` in the `primary_response`.
We are not taking any action as of now if the void fails but the response is sent to the active-merchant-service.

Reply:
is this an abstraction that active merchant provides? I thought this was provided by dlocal itself, no?

Reply:
No, dlocal only allows us to save card during the authorization charge API call. Active-merchant does the void call immediately after the authorize call is successful.

Reply:
Okay got it.